bagit.py

```diff
@@ -162,30 +163,41 @@ def make_bag(
     if checksums is None:
         checksums = DEFAULT_CHECKSUMS
 
-    bag_dir = os.path.abspath(bag_dir)
+    if dest_dir:
+        bag_name = os.path.basename(bag_dir)
+        dest_dir = os.path.abspath(os.path.join(dest_dir, bag_name))
+        if not os.path.isdir(dest_dir):
+            os.makedirs(dest_dir)
+        else:
+            raise RuntimeError(_("The following directory already exists:\n%s"), dest_dir)
+    else:
+        dest_dir = os.path.abspath(bag_dir)
+
+    source_dir = os.path.abspath(bag_dir)
```

Comment:
Consider using `realpath` instead to resolve symlinks (not related to this change, just a general observation)

Reply:
I couldn't think of any edge cases, but I did have to invoke realpath in the setup class for the tests.